src/main/java/au/gov/ga/geodesy/support/marshalling/moxy/GeodesyMLMoxy.java

```diff
@@ -82,6 +83,25 @@ public void marshal(JAXBElement<?> site, Writer writer) throws MarshallingExcept
         }
     }
 
+    public void marshal(Object x, Writer writer) throws MarshallingException {
+        // TODO: how can we restrict x?
+        String typeName = x.getClass().getSimpleName();
+        String factoryName = x.getClass().getPackage().getName() + ".ObjectFactory";
+        String factoryMethodName = "create" + typeName.substring(0, typeName.length() - "Type".length());
+        try {
+            Class<?> factoryClass = Class.forName(factoryName);
+            Object factory = factoryClass.newInstance();
+            Method factoryMethod = factoryClass.getMethod(factoryMethodName, new Class<?>[]{x.getClass()});
+            JAXBElement<?> element = (JAXBElement<?>) factoryMethod.invoke(factory, new Object[]{x});
```

Comment:
If the factory method does not return a JAXBElement, then the cast to JAXBElement will fail with an exception. If we do the check for the return type and get something other than a JAXBElement, we would presumable throw an exception. Either way, we can't recover. Let's see how this code fares in practice.